Src/FluentAssertions/Common/StringExtensions.cs

```diff
@@ -41,6 +42,14 @@ public static string IndexedSegmentAt(this string value, int index)
             return $"{formattedString} (index {index})".EscapePlaceholders();
         }
 
+        /// <summary>
+        /// Replaces the numeric index from a path like "property[0].nested" and returns "property[].nested"
+        /// </summary>
+        public static string WithoutSpecificCollectionIndex(this string indexedPath)
```

Comment:
This method removes all indices, and not just a single index.
```suggestion
        /// <summary>
        /// Replaces the numeric indices from a path like "property[0].nested" and returns "property[].nested"
        /// </summary>
        public static string WithoutSpecificCollectionIndicesthis string indexedPath)
```